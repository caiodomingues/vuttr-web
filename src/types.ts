export type Tag = {
  name: string;
};

export type Tool = {
  id: number;
  title: string;
  description: string;
  link: string;
  tags: Array<string>;
};
