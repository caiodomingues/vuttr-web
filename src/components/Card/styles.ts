import styled from "styled-components";

export const Container = styled.div`
  background: #ffffff 0% 0% no-repeat padding-box;
  box-shadow: 0px 5px 7px #0000000d;
  border: 1px solid #ebeaed;
  border-radius: 5px;

  margin: 16px 0;
  padding: 16px;

  div {
    display: flex;
    flex-direction: row;
    justify-content: space-between;

    a {
      text-align: left;
      font: normal normal 600 30px/36px "Source Sans Pro";
      color: #365df0;
      letter-spacing: 0.6px;

      @media screen and (max-width: 525px) {
        font-size: 24px;
        line-height: 30px;
      }
    }

    button {
      text-align: center;
      font: normal normal 600 18px/24px "Source Sans Pro";
      letter-spacing: 0.36px;
      color: #f95e5a;
      background-color: inherit;
      border: none;

      cursor: pointer;

      &:disabled {
        color: #fcaeac;
        cursor: not-allowed;
      }
    }

    .remove {
      background: transparent 0% 0% no-repeat padding-box;
      width: 10px;
      height: 10px;
    }
  }

  span {
    margin-right: 8px;
    font-weight: 600;
  }
`;
