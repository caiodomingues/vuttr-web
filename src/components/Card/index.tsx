import { ReactNode } from "react";

import { Tool } from "../../types";
import { Container } from "./styles";

import { ReactComponent as Close } from "../../assets/img/icon-close.svg";

interface CardProps {
  tool: Tool;
  handleDeleteTool: Function;
  children?: ReactNode;
}

function Card({ tool, handleDeleteTool, children }: CardProps) {
  const { id, title, description, link, tags } = tool;

  return (
    <Container>
      <div>
        <a target="_blank" rel="noreferrer" href={link}>
          {title}
        </a>
        <button onClick={() => handleDeleteTool(id)}>
          <Close className="remove" /> remove
        </button>
      </div>
      <p>{description}</p>
      {tags.map((tag, index) => (
        <span key={index}>#{tag}</span>
      ))}
    </Container>
  );
}

export default Card;
