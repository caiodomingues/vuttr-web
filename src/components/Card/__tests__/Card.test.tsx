// Link.react.test.js
import renderer from "react-test-renderer";
import Card from "..";
import { Tool } from "../../../types";

const MockTool: Tool = {
  id: 0,
  title: "Mock tool",
  link: "http://localhost:8000",
  description: "Lorem ipsum dolor sit amet.",
  tags: ["php"],
};

const handleDeleteTool = (id: number) => {
  //
};

test("Should render a card", () => {
  const component = renderer.create(
    <Card tool={MockTool} handleDeleteTool={handleDeleteTool} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
