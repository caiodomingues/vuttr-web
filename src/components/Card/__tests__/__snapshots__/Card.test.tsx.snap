// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a card 1`] = `
<div
  className="sc-bdnxRM gATyKY"
>
  <div>
    <a
      href="http://localhost:8000"
      rel="noreferrer"
      target="_blank"
    >
      Mock tool
    </a>
    <button
      onClick={[Function]}
    >
      <svg
        className="red"
      >
        icon-close.svg
      </svg>
       remove
    </button>
  </div>
  <p>
    Lorem ipsum dolor sit amet.
  </p>
  <span>
    #
    php
  </span>
</div>
`;

exports[`Should render a modal 1`] = `
<div
  className="sc-bdnxRM gATyKY"
>
  <div>
    <a
      href="http://localhost:8000"
      rel="noreferrer"
      target="_blank"
    >
      Mock tool
    </a>
    <button
      onClick={[Function]}
    >
      <svg
        className="red"
      >
        icon-close.svg
      </svg>
       remove
    </button>
  </div>
  <p>
    Lorem ipsum dolor sit amet.
  </p>
  <span>
    #
    php
  </span>
</div>
`;
